From 1f07f79b0384396846276d194d80bd83c14c966e Mon Sep 17 00:00:00 2001
From: Huw Jones <huw@pexip.com>
Date: Fri, 4 Feb 2022 14:40:18 +0000
Subject: [PATCH] access log: add (P)ID without <>

Allows outputting an access log in the same format as the web server log.
---
 gunicorn/config.py   | 3 ++-
 gunicorn/glogging.py | 1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

Index: gunicorn/gunicorn/config.py
===================================================================
--- gunicorn.orig/gunicorn/config.py
+++ gunicorn/gunicorn/config.py
@@ -1418,7 +1418,8 @@ class AccessLogFormat(Setting):
         M            request time in milliseconds
         D            request time in microseconds
         L            request time in decimal seconds
-        p            process ID
+        P            process ID
+        p            <process ID>
         {header}i    request header
         {header}o    response header
         {variable}e  environment variable
Index: gunicorn/gunicorn/glogging.py
===================================================================
--- gunicorn.orig/gunicorn/glogging.py
+++ gunicorn/gunicorn/glogging.py
@@ -314,6 +314,7 @@ class Logger:
             'D': (request_time.seconds * 1000000) + request_time.microseconds,
             'M': (request_time.seconds * 1000) + int(request_time.microseconds / 1000),
             'L': "%d.%06d" % (request_time.seconds, request_time.microseconds),
+            'P': str(os.getpid()),
             'p': "<%s>" % os.getpid()
         }
 
